app.ts

```diff
@@ -0,0 +1,1160 @@
+	// intercept logs and redirect them to socket
+	const interceptor = function (
+		write: (buffer: string | Uint8Array, cb?: (err?: Error) => void) => void
+	) {
+		return function (...args: any[]): boolean {
+			socketManager.io.emit('DEBUG', args[0]?.toString())
+			return write.apply(process.stdout, args)
+		}
+	}
+
+	// eslint-disable-next-line @typescript-eslint/unbound-method
+	process.stdout.write = interceptor(process.stdout.write)
+	// eslint-disable-next-line @typescript-eslint/unbound-method
+	process.stderr.write = interceptor(process.stderr.write)
+}
+
+async function parseDir(dir: string | boolean): Promise<StoreFileEntry[]> {
+	const toReturn = []
+	const files = await fs.readdir(dir)
+	for (const file of files) {
+		const entry: StoreFileEntry = {
+			name: path.basename(file),
+			path: utils.joinPath(dir, file),
+		}
+		const stats = await fs.lstat(entry.path)
+		if (stats.isDirectory()) {
+			if (entry.name === '.config-db') {
+				// hide config-db
+				continue
+			}
+			entry.children = await parseDir(entry.path)
+		} else {
+			entry.ext = file.split('.').pop()
+		}
+
+		entry.size = utils.humanSize(stats.size)
+		toReturn.push(entry)
+	}
+	return toReturn
+}
+
+// ### EXPRESS SETUP
+
+logger.info(`Version: ${utils.getVersion()}`)
+logger.info('Application path:' + utils.getPath(true))
+
+// view engine setup
+app.set('views', utils.joinPath(false, 'views'))
+app.set('view engine', 'ejs')
+
+app.use(
+	morgan('dev', {
+		stream: { write: (msg: string) => logger.info(msg.trimEnd()) },
+	})
+)
+app.use(express.json({ limit: '50mb' }))
+app.use(
+	express.urlencoded({
+		limit: '50mb',
+		extended: true,
+		parameterLimit: 50000,
+	})
+)
+
+app.use(
+	history({
+		index: '/',
+	})
+)
+
+app.get('/', apisLimiter, renderIndex)
+
+app.use('/', express.static(utils.joinPath(false, 'dist')))
+
+app.use(cors({ credentials: true, origin: true }))
+
+// enable sessions management
+app.use(
+	session({
+		name: 'zwavejs2mqtt-session',
+		secret: sessionSecret,
+		resave: false,
+		saveUninitialized: false,
+		store: new FileStore({
+			path: path.join(storeDir, 'sessions'),
+			logFn: (...args: any[]) => {
+				// skip ENOENT errors
+				if (
+					args &&
+					args.filter((a) => a.indexOf('ENOENT') >= 0).length === 0
+				) {
+					logger.debug(args[0])
+				}
+			},
+		}),
+		cookie: {
+			secure: !!process.env.HTTPS || !!process.env.USE_SECURE_COOKIE,
+			httpOnly: true, // prevents cookie to be sent by client javascript
+			maxAge: 24 * 60 * 60 * 1000, // one day
+		},
+	})
+)
+
+// Node.js CSRF protection middleware.
+// Requires either a session middleware or cookie-parser to be initialized first.
+const csrfProtection = csrf({
+	value: (req) => req.csrfToken(),
+})
+
+// ### SOCKET SETUP
+
+/**
+ * Binds socketManager to `server`
+ *
+ * @param {HttpServer} server
+ */
+function setupSocket(server: HttpServer) {
+	socketManager.bindServer(server)
+
+	socketManager.on(inboundEvents.init, function (socket: Socket) {
```

Comment:
try to avoid inline function type declarations - let TS infer them from the method signature that has the callback. (Example see Slack)

Reply:
Fixed